### ISSUE TITLE
ppo_mini_batch_size for Non-Cumulative Agents

### ISSUE THREAD

**Issue by huseyinatahaninan** [rllm-org/rllm#350]
If my setting is based on non-cumulative agents, trajectories may vary in length (e.g., one with 10 steps, another with 8) so train_batch_size is expanded into a size that cannot be predetermined, it will essentially be the total number of steps over all trajectories within the rollouts. In that case, how can one even set ppo_mini_batch_size so that the training is on-policy in GRPO let's say? This is not possible because the batch size is dynamic, what's the right way to handle this situation?

**Comment by LianShuQuan** [comment]
train_batch_size and ppo_mini_batch_size are both prompt level (how many task a iteration have) . 
xxx_batch_size_per_gpu is actual sequence num in gpu since there exist the gradient accumulation. (You may also use sequence parallel)
See #https://github.com/volcengine/verl/issues/1524

**Comment by huseyinatahaninan** [comment]
Appreciate your comment! I think my setting is different in the sense that for cumulative agents, one prompt would give one trajectory but as I use non-cumulative setting, one prompt can give varying number of trajectories. I guess I need to find the way to make the implementation such that all the trajectories for a single prompt are packed together so that ppo_mini_batch_size stays at prompt level as well. I'll try to find some similar settings out there and see how folks implement this. I'd appreciate any pointers from anyone though.
